test/python/test_prod.py

```diff
@@ -42,51 +41,47 @@ def test_prod(self, v1, axis, expected):
         tensor = tf.placeholder(tf.float32, shape=(None))
         assert np.allclose(np.prod(v1, axis), expected)
         out = tf.reduce_prod(tensor, axis=axis)
-        sess_fn = sess.run([out], feed_dict={tensor: v1})
+        sess_fn = lambda sess: sess.run([out], feed_dict={tensor: v1})
         assert np.allclose(
             self.with_ngraph(sess_fn), self.without_ngraph(sess_fn))
 
     @pytest.mark.parametrize(("v1", "expected"), (((2.0, 2.0), [4.0]),))
     def test_prod_no_axis(self, v1, expected):
         tensor = tf.placeholder(tf.float32, shape=(None))
-
-        with self.device:
-            out = tf.reduce_prod(tensor)
-            cfg = self.config
-            # Rank op is currently not supported by NGRAPH
-            cfg.allow_soft_placement = True
-            with tf.Session(config=cfg) as sess:
-                result = sess.run((out,), feed_dict={tensor: v1})
-                assert np.allclose(result, expected)
+        out = tf.reduce_prod(tensor)
+        # Rank op is currently not supported by NGRAPH
+        cfg = tf.ConfigProto(allow_soft_placement=True)
+        with tf.Session(config=cfg) as sess:
```

Comment:
`cfg = tf.ConfigProto(allow_soft_placement=True)` & `with tf.Session(config=cfg) as sess:`: these lines are not needed i think